### PR TITLE
fix(plugin): fix legacy _register_controls

### PR DESCRIPTION
**Pull request by crazypsycho**

### CODE REVIEW

File: trunk/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "requires": true,
   "lockfileVersion": 1,
   "dependencies": {
-    "gulp": "^3.9.1",
+    "gulp": "^4.0.2",
```
Comment (flammehawk) [inline trunk/package.json:5]: is this update tested?

**Review by flammehawk — APPROVED** [review]

**Comment by eac0n** [comment]
I would have had an objection.
All these
```php
protected function _register_controls() {
    $this->register_controls();
}
```
should be part of a common abstract parent class to avoid that every single class implements exactly the same method again.

**Comment by crazypsycho** [comment]
I will have a look tomorrow

**Comment by crazypsycho** [comment]
The classes already extend different classes from elementor like Tag and DataTag (in our internal plugin its also Widget_Base).
So we would need more than one abstract base-class. I would prefere a trait for this.

**Comment by eac0n** [comment]
Ok, use a trait then.

**Pull request merged**